hbase/src/main/java/org/apache/metamodel/hbase/HBaseClient.java

```diff
@@ -0,0 +1,169 @@
+ */
+public final class HBaseClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HBaseClient.class);
+
+    private final Connection _connection;
+
+    public HBaseClient(Connection connection) {
+        this._connection = connection;
+    }
+
+    /**
+     * Insert a single row of values to a HBase table.
+     * @param tableName
+     * @param columns
+     * @param values
+     * @throws IllegalArgumentException when any parameter is null or the indexOfIdColumn is impossible
+     * @throws MetaModelException when no ID-column is found.
+     * @throws MetaModelException when a {@link IOException} is catched
+     */
+    // TODO: Use the ColumnTypes to determine the inserts. Now the toString() method is called on the object.
+    public void insertRow(String tableName, HBaseColumn[] columns, Object[] values, int indexOfIdColumn) {
+        if (tableName == null || columns == null || values == null || indexOfIdColumn >= values.length
+                || values[indexOfIdColumn] == null) {
+            throw new IllegalArgumentException(
+                    "Can't insert a row without having (correct) tableName, columns, values or indexOfIdColumn");
+        }
+        try (final Table table = _connection.getTable(TableName.valueOf(tableName))) {
+            // Create a put with the values of indexOfIdColumn as rowkey
+            final Put put = new Put(Bytes.toBytes(values[indexOfIdColumn].toString()));
+
+            // Add the other values to the put
+            for (int i = 0; i < columns.length; i++) {
+                if (i != indexOfIdColumn) {
+                    put.addColumn(Bytes.toBytes(columns[i].getColumnFamily()), Bytes.toBytes(columns[i].getQualifier()),
+                            Bytes.toBytes(values[i].toString()));
+                }
+            }
+            // Add the put to the table
+            table.put(put);
+        } catch (IOException e) {
+            throw new MetaModelException(e);
+        }
+    }
+
+    /**
+     * Delete 1 row based on the key
+     * @param tableName
+     * @param rowKey
+     * @throws IllegalArgumentException when any parameter is null
+     * @throws MetaModelException when a {@link IOException} is catched
+     */
+    public void deleteRow(String tableName, Object rowKey) {
+        if (tableName == null || rowKey == null) {
+            throw new IllegalArgumentException("Can't delete a row without having tableName or rowKey");
+        }
+        byte[] rowKeyAsByteArray = Bytes.toBytes(rowKey.toString());
+        if (rowKeyAsByteArray.length > 0) {
+            try (final Table table = _connection.getTable(TableName.valueOf(tableName));) {
+                if (rowExists(table, rowKeyAsByteArray) == true) {
+                    table.delete(new Delete(rowKeyAsByteArray));
+                } else {
+                    logger.warn("Rowkey with value " + rowKey.toString() + " doesn't exist in the table");
```

Comment:
For performance reasons it's better to use a Marker object when inserting variables into log messages, e.g.:
```
logger.warn("Rowkey with value {} doesn't exist in the table", rowKey.toString());
```

Reply:
Can you give me something to read on this (why it's needed and how to implement it)?

Reply:
See https://rules.sonarsource.com/java/RSPEC-2629.